### DISCUSSION TITLE
Can't install leantime properly

### DISCUSSION THREAD

**Discussion by ghost** [Leantime/leantime#553] [Q&A] [answered]
Sorry to bother but i've spent like 10h trying to make this work and i cant.
So i've got a raspberry that i wiped and updated to raspberry os (debian fork).
I installed the following :

- PHP 8.1.1
- Mysql Ver 15.1 Distrib 10.5.12-MariaDB, for debian-linux-gnueabihf (armv7l)
- Apache/2.4.52 (Raspbian) (fresh install no config)
- libreapache-mod-php
- php-mysql

From what i've read online, i now need to extract the release archive into `var/www/html/leantime/` (because i have phpmyadmin as another subsite)

So i now have` html/leantime/public/` in my hierarchy.
I rename the config sample file and enter my database info :
```php
  /* General */
  public $sitename = "Leantime"; //Name of your site, can be changed later
  public $language = "en-US"; //Default language
  public $mainColor = "1b75bb"; //Default color, can be changed later
  public $logoPath = "/images/logo.png"; //Default logo path, can be changed later
  public $appUrl = "localhost/leantime"; //<--- is this ok ? i'll access the site through 192.168.0.14/leantime

  /* Database */
  public $dbHost="localhost"; //Database host
  public $dbUser="leantime"; //Database username
  public $dbPassword="MYPASSWORD"; //Database password
  public $dbDatabase="leantime"; //Database name
```
i then open `/html/leantime/public/.htaccess` and uncomment `RewriteBase /leantime/`

Ok so now i need to allow htaccess file into apache2 right ?
I open `etc/apache2/apache2.conf` and edit to :
```xml
<Directory /var/www/>
	Options Indexes FollowSymLinks
	AllowOverride All
	Require all granted
</Directory>
```
Finally i do `sudo a2enmod rewrite` and `systemctl restart apache2`

But if i try and open `http://192.168.0.14/leantime/` i get some php code in the page :
![image](https://user-images.githubusercontent.com/2307228/148208094-7b1fec10-b8f2-42c3-9813-35a774f1cf47.png)

**Comment by DFF-fred** [comment]
Hi!
Where did you get leantime from? I'd say most likely you downloaded the code from github instead of the compiled version. Some folders are missing and it doesn't work. I also don´t think you had to edit the .htaccess (I didn't ) but.. it's another topic.
You can find the compiled versions (and source) links here: https://github.com/Leantime/leantime/releases
Then if this doesn't work, you also have an error log file in the log directory.

**Comment by ghost** [comment, marked as answer]
No i downloaded the release version on leantime.io
For the php message i forgot to enable apache2 php module which is good now.
I fixed it by doing 

- sudo apt purge libapache2-mod-php8.1 libapache2-mod-php
- sudo apt-get install libapache2-mod-php8.1
- sudo a2enmod php8.1

And then restarted everything.
Also the app url need the 192.168.0.14 instead of localhost

Thank you for your help :) Everything works finally

**Reply by ghost** [reply]
However, the installation page show me this after i type in a login 
![image](https://user-images.githubusercontent.com/2307228/148217775-a37cc1e2-6405-41e1-b6cb-0e43c303ab58.png)
Nothing is moving. I do see the DB is being populated though.

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/2307228/148218004-e7ca8df4-feab-48ab-b064-4ed933798af3.png)
After login i get a blank page with this error in ressource/error.log 
```txt
[05-Jan-2022 04:25:16 America/Los_Angeles] PHP Fatal error:  Uncaught PDOException: There is no active transaction in /var/www/html/leantime/src/core/class.install.php:181
Stack trace:
#0 /var/www/html/leantime/src/core/class.install.php(181): PDO->rollBack()
#1 /var/www/html/leantime/src/install.php(105): leantime\core\install->setupDB()
#2 /var/www/html/leantime/src/core/class.application.php(69): include('...')
#3 /var/www/html/leantime/public/index.php(41): leantime\core\application->start()
#4 {main}
  thrown in /var/www/html/leantime/src/core/class.install.php on line 181
[05-Jan-2022 04:26:50 America/Los_Angeles] PHP Deprecated:  explode(): Passing null to parameter #2 ($string) of type string is deprecated in /var/www/html/leantime/src/core/class.session.php on line 52
[05-Jan-2022 04:30:38 America/Los_Angeles] PHP Deprecated:  unserialize(): Passing null to parameter #1 ($data) of type string is deprecated in /var/www/html/leantime/src/core/class.login.php on line 476
[05-Jan-2022 04:30:39 America/Los_Angeles] PHP Deprecated:  Optional parameter $projectId declared before required parameter $userId is implicitly treated as a required parameter in /var/www/html/leantime/src/domain/timesheets/repositories/class.timesheets.php on line 229
[05-Jan-2022 04:30:39 America/Los_Angeles] PHP Deprecated:  Optional parameter $dateStart declared before required parameter $userId is implicitly treated as a required parameter in /var/www/html/leantime/src/domain/timesheets/repositories/class.timesheets.php on line 229
[05-Jan-2022 04:30:39 America/Los_Angeles] PHP Fatal error:  Uncaught TypeError: PDOStatement::setFetchMode(): Argument #2 must be a valid class in /var/www/html/leantime/src/domain/reports/repositories/class.reports.php:169
Stack trace:
#0 /var/www/html/leantime/src/domain/reports/repositories/class.reports.php(169): PDOStatement->setFetchMode()
#1 /var/www/html/leantime/src/domain/reports/services/class.reports.php(33): leantime\domain\repositories\reports->checkLastReportEntries()
#2 /var/www/html/leantime/src/domain/dashboard/controllers/class.show.php(35): leantime\domain\services\reports->dailyIngestion()
#3 /var/www/html/leantime/src/core/class.frontcontroller.php(148): leantime\domain\controllers\show->__construct()
#4 /var/www/html/leantime/src/core/class.frontcontroller.php(107): leantime\core\frontcontroller->executeAction()
#5 /var/www/html/leantime/src/core/class.application.php(91): leantime\core\frontcontroller->run()
#6 /var/www/html/leantime/public/index.php(41): leantime\core\application->start()
#7 {main}
  thrown in /var/www/html/leantime/src/domain/reports/repositories/class.reports.php on line 169

```

**Reply by ghost** [reply]
It seems leantime is not yet compatible with php8. i need to downgrade to php7.2..
